<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>769</width>
    <height>636</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>NCKU_Project_2</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTableWidget" name="tableWidget">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>40</y>
      <width>542</width>
      <height>542</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">font: 30pt &quot;Ubuntu&quot;;</string>
    </property>
    <property name="showGrid">
     <bool>true</bool>
    </property>
    <property name="gridStyle">
     <enum>Qt::CustomDashLine</enum>
    </property>
    <property name="rowCount">
     <number>9</number>
    </property>
    <property name="columnCount">
     <number>9</number>
    </property>
    <attribute name="horizontalHeaderVisible">
     <bool>false</bool>
    </attribute>
    <attribute name="horizontalHeaderDefaultSectionSize">
     <number>60</number>
    </attribute>
    <attribute name="verticalHeaderVisible">
     <bool>false</bool>
    </attribute>
    <attribute name="verticalHeaderDefaultSectionSize">
     <number>60</number>
    </attribute>
    <attribute name="verticalHeaderHighlightSections">
     <bool>true</bool>
    </attribute>
    <row/>
    <row/>
    <row/>
    <row/>
    <row/>
    <row/>
    <row/>
    <row/>
    <row/>
    <column/>
    <column/>
    <column/>
    <column/>
    <column/>
    <column/>
    <column/>
    <column/>
    <column/>
    <item row="0" column="0">
     <property name="text">
      <string/>
     </property>
     <property name="textAlignment">
      <set>AlignLeading|AlignVCenter</set>
     </property>
    </item>
    <item row="0" column="1">
     <property name="text">
      <string/>
     </property>
     <property name="textAlignment">
      <set>AlignLeading|AlignVCenter</set>
     </property>
    </item>
    <item row="0" column="2">
     <property name="text">
      <string/>
     </property>
     <property name="textAlignment">
      <set>AlignLeading|AlignVCenter</set>
     </property>
    </item>
    <item row="0" column="3">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="0" column="4">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="0" column="5">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="0" column="6">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="0" column="7">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="0" column="8">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="1" column="0">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="1" column="1">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="1" column="2">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="1" column="3">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="1" column="4">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="1" column="5">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="1" column="6">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="1" column="7">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="1" column="8">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="2" column="0">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="2" column="1">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="2" column="2">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="2" column="3">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="2" column="4">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="2" column="5">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="2" column="6">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="2" column="7">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="2" column="8">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="3" column="0">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="3" column="1">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="3" column="2">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="3" column="3">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="3" column="4">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="3" column="5">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="3" column="6">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="3" column="7">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="3" column="8">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="4" column="0">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="4" column="1">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="4" column="2">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="4" column="3">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="4" column="4">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="4" column="5">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="4" column="6">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="4" column="7">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="4" column="8">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="5" column="0">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="5" column="1">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="5" column="2">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="5" column="3">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="5" column="4">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="5" column="5">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="5" column="6">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="5" column="7">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="5" column="8">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="6" column="0">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="6" column="1">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="6" column="2">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="6" column="3">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="6" column="4">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="6" column="5">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="6" column="6">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="6" column="7">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="6" column="8">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="7" column="0">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="7" column="1">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="7" column="2">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="7" column="3">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="7" column="4">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="7" column="5">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="7" column="6">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="7" column="7">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="7" column="8">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="8" column="0">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="8" column="1">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="8" column="2">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="8" column="3">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="8" column="4">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="8" column="5">
     <property name="text">
      <string/>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>211</red>
        <green>215</green>
        <blue>207</blue>
       </color>
      </brush>
     </property>
    </item>
    <item row="8" column="6">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="8" column="7">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="8" column="8">
     <property name="text">
      <string/>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="Clock">
    <property name="geometry">
     <rect>
      <x>613</x>
      <y>40</y>
      <width>131</width>
      <height>91</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>00 : 00 : 00</string>
    </property>
   </widget>
   <widget class="Line" name="line_5">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>40</y>
      <width>3</width>
      <height>541</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <weight>50</weight>
      <bold>false</bold>
     </font>
    </property>
    <property name="lineWidth">
     <number>2</number>
    </property>
    <property name="midLineWidth">
     <number>10</number>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="Line" name="line_7">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>40</y>
      <width>3</width>
      <height>541</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <weight>50</weight>
      <bold>false</bold>
     </font>
    </property>
    <property name="lineWidth">
     <number>2</number>
    </property>
    <property name="midLineWidth">
     <number>10</number>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="Line" name="line_8">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>220</y>
      <width>541</width>
      <height>3</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <weight>50</weight>
      <bold>false</bold>
     </font>
    </property>
    <property name="lineWidth">
     <number>2</number>
    </property>
    <property name="midLineWidth">
     <number>10</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_9">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>400</y>
      <width>541</width>
      <height>3</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <weight>50</weight>
      <bold>false</bold>
     </font>
    </property>
    <property name="lineWidth">
     <number>2</number>
    </property>
    <property name="midLineWidth">
     <number>10</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>340</y>
      <width>126</width>
      <height>241</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QPushButton" name="pause">
       <property name="autoFillBackground">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Pause</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="Undo">
       <property name="text">
        <string>Undo</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="NewPuzzle">
       <property name="text">
        <string>Start New Puzzle</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="resultLabel">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>160</y>
      <width>191</width>
      <height>91</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>30</pointsize>
      <weight>75</weight>
      <italic>true</italic>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Bingo</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>769</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuUser">
    <property name="title">
     <string>Game</string>
    </property>
    <widget class="QMenu" name="menuNew_Puzzle">
     <property name="title">
      <string>Start New Puzzle</string>
     </property>
     <addaction name="actionsetRandomly"/>
     <addaction name="actionsetManually"/>
    </widget>
    <addaction name="separator"/>
    <addaction name="menuNew_Puzzle"/>
    <addaction name="actionCheck"/>
    <addaction name="actionReset"/>
   </widget>
   <widget class="QMenu" name="menuHint">
    <property name="title">
     <string>Hint</string>
    </property>
    <addaction name="actionGet_Hint"/>
    <addaction name="actionSolve"/>
   </widget>
   <addaction name="menuUser"/>
   <addaction name="menuHint"/>
  </widget>
  <action name="actionCheck">
   <property name="text">
    <string>Finish</string>
   </property>
  </action>
  <action name="actionsetRandomly">
   <property name="text">
    <string>Random Puzzle</string>
   </property>
  </action>
  <action name="actionsetManually">
   <property name="text">
    <string>Manual Puzzle</string>
   </property>
  </action>
  <action name="actionSolve">
   <property name="text">
    <string>Solve</string>
   </property>
  </action>
  <action name="actionReset">
   <property name="text">
    <string>Reset</string>
   </property>
  </action>
  <action name="actionGet_Hint">
   <property name="text">
    <string>Get Hint</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
